### PR TITLE
Added `pairs` function to Collection 

### PR DESCRIPTION
**Pull request by boudhayan**
Added a new function `pairs` to the collection extension to generate unique pairs of elements.
## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 5.0.
- [x] New extensions support iOS 10.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+, or use `@available` if not.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

**Review by LucianoPAlmeida — COMMENTED** [review]
Thank you for the contribution, left a couple small comments.

File: Tests/SwiftStdlibTests/CollectionExtensionsTests.swift [hunk 1]
```diff
@@ -141,4 +141,18 @@ final class CollectionExtensionsTests: XCTestCase {
         XCTAssertEqual([1, 2, 3, 4, 5].average(), 3)
         XCTAssertEqual([Int]().average(), 0)
     }
+    
+    func testPairs() {
```
[the remark this answers is not in the document]
Comment (boudhayan) [reply Tests/SwiftStdlibTests/CollectionExtensionsTests.swift]: @LucianoPAlmeida I have added tests for collection with even, odd numbers of elements and empty collection.
Could you please check?

File: Tests/SwiftStdlibTests/CollectionExtensionsTests.swift [hunk 2]
```diff
@@ -141,4 +141,37 @@ final class CollectionExtensionsTests: XCTestCase {
         XCTAssertEqual([1, 2, 3, 4, 5].average(), 3)
         XCTAssertEqual([Int]().average(), 0)
     }
+    
+    func testAdjacentPairsWithOddNumberOfElementsInCollection() {
+        var pairs: [(Int, Int)] = []
+        for (first, second) in [1, 2, 3].adjacentPairs() {
+            pairs.append((first, second))
+        }
```
Comment (LucianoPAlmeida) [inline Tests/SwiftStdlibTests/CollectionExtensionsTests.swift:149]: Nit: Would that be simpler?
```suggestion
        let pairs = Array([1, 2, 3].adjacentPairs())  
```

Comment (LucianoPAlmeida) [reply Tests/SwiftStdlibTests/CollectionExtensionsTests.swift:149]: Valid for other occurrences as well if applicable.

File: Sources/SwifterSwift/SwiftStdlib/CollectionExtensions.swift [hunk 3]
```diff
@@ -87,6 +87,39 @@ public extension Collection {
...
+    
+    /// SwifterSwift: Unique pair of elements in a collection.
+    ///
+    ///        let array = [1, 2, 3]
+    ///        for (first, second) in array.pairs() {
+    ///            print(first, second) // print: (1, 2) (1, 3) (2, 3)
+    ///        }
+    ///
+    ///
+    /// - Returns: a sequence with tuple(unique pair) as Element type.
```
Comment (LucianoPAlmeida) [inline Sources/SwifterSwift/SwiftStdlib/CollectionExtensions.swift:99]:
```suggestion
    /// - Returns: a sequence of adjacent pairs of elements from this collection.
```

File: Tests/SwiftStdlibTests/CollectionExtensionsTests.swift (hunk 2 above)
Comment (boudhayan) [reply Tests/SwiftStdlibTests/CollectionExtensionsTests.swift]: Done. 
replaced with the above kind of code wherever applicable.
Thanks for the suggestion!

**Review by LucianoPAlmeida — APPROVED** [review]

**Pull request merged**